File: Atividades/atvd4/haskell/Concatena.hs
module Concatena (concatena) where

-- concatena :: [a]->[a]->[a]
concatena [] [] = []
concatena xs [] = xs
concatena (x:xs) (y:ys)
    | xs == [] = x:ys
    |otherwise = x:(concatena (xs) (y:ys))

File: Atividades/atvd4/javascript/Concatena.js
function concatena(xs,ys){
    return 'error'
    
    if(xs ==[] && ys==[]) return []
    else if( xs.length>0 && ys==[] ) return xs
    else if(ys.length >0 && xs ==[])   return ys
    else{
        return [xs[0],...concatena(xs,ys.slice(1)) ]
    }
    
}


module.exports = concatena